shared/diagram.py

```diff
@@ -45,14 +133,102 @@ def make_directories():
             else:
                 print("Successfully created the directory %s " % PATH_DIAGRAM_OUTPUT)
 
+    def group_by_group(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/gh/Cloud-Architects/cloud-discovery/pullRequest?prid=5655786)